packages/flutter/lib/src/material/segmented_button.dart

```diff
@@ -178,6 +181,127 @@ class SegmentedButton<T> extends StatefulWidget {
+  /// [ButtonStyle] given simple values.
+  ///
+  /// The [selectedForegroundColor] color is used to create a
+  /// [MaterialStateProperty] [ButtonStyle.foregroundColor].
+  ///
+  /// The [selectedBackgroundColor] color is used to create a
+  /// [MaterialStateProperty] [ButtonStyle.backgroundColor].
+  ///
+  /// The [foregroundColor] and [disabledForegroundColor] colors are used
+  /// to create a [MaterialStateProperty] [ButtonStyle.foregroundColor], and
+  /// a derived [ButtonStyle.overlayColor].
+  ///
+  /// The [backgroundColor] and [disabledBackgroundColor] colors are
+  /// used to create a [MaterialStateProperty] [ButtonStyle.backgroundColor].
+  ///
+  /// Similarly, the [enabledMouseCursor] and [disabledMouseCursor]
+  /// parameters are used to construct [ButtonStyle.mouseCursor].
+  ///
+  /// All of the other parameters are either used directly or used to
+  /// create a [MaterialStateProperty] with a single value for all
+  /// states.
+  ///
+  /// All parameters default to null. By default this method returns
+  /// a [ButtonStyle] that doesn't override anything.
+  ///
+  /// For example, to override the default text and icon colors for a
+  /// [SegmentedButton], as well as its overlay color, with all of the
+  /// standard opacity adjustments for the pressed, focused, and
+  /// hovered states, one could write:
+  ///
+  /// ```dart
+  /// SegmentedButton<int>(
+  ///   style: SegmentedButton.styleFrom(foregroundColor: Colors.green),
+  ///   segments: const <ButtonSegment<int>>[
+  ///      ButtonSegment<int>(
+  ///          value: 0, label: Text('0'), icon: Icon(Icons.calendar_view_day)),
+  ///      ButtonSegment<int>(
+  ///          value: 1, label: Text('1'), icon: Icon(Icons.calendar_view_week)),
+  ///      ButtonSegment<int>(
+  ///          value: 2, label: Text('2'), icon: Icon(Icons.calendar_view_month)),
+  ///      ButtonSegment<int>(
+  ///          value: 3, label: Text('3'), icon: Icon(Icons.calendar_today)),
+  ///    ],
+  ///    selected: const <int>{0},
+  ///    onSelectionChanged: (Set<int> selection) {},
+  /// ),
+  /// ```
+  static ButtonStyle styleFrom({
+    Color? foregroundColor,
+    Color? selectedForegroundColor,
+    Color? selectedBackgroundColor,
+    Color? backgroundColor,
+    Color? disabledForegroundColor,
+    Color? disabledBackgroundColor,
+    Color? shadowColor,
+    Color? surfaceTintColor,
+    double? elevation,
+    TextStyle? textStyle,
+    EdgeInsetsGeometry? padding,
+    Size? minimumSize,
+    Size? fixedSize,
+    Size? maximumSize,
+    BorderSide? side,
+    OutlinedBorder? shape,
+    MouseCursor? enabledMouseCursor,
+    MouseCursor? disabledMouseCursor,
+    VisualDensity? visualDensity,
+    MaterialTapTargetSize? tapTargetSize,
+    Duration? animationDuration,
+    bool? enableFeedback,
+    AlignmentGeometry? alignment,
+    InteractiveInkFeatureFactory? splashFactory,
+    Color? primary,
+    Color? onPrimary,
+    Color? onSurface,
```

Comment:
```suggestion

```

These color properties are deprecated in other button style buttons so we shouldn't add them here.

Reply:
firstly, l added @deprecated message for these properties. but l took error from Cİ l could not pass tests. 
Should add this messages because other buttons still have these properties or should l remove primary, onPrimary, onSurface.
![image](https://github.com/flutter/flutter/assets/65075121/1605b5ad-7734-4bf2-974c-9b09833fe880)

Reply:
Deprecated parameters is only for existing class.  it shouldn't added to this class. Please follow the suggestions